airflow/cli/commands/connection_command.py

```diff
@@ -67,6 +69,52 @@ def connections_list(args):
         print(msg)
 
 
+def _format_connections(conns: List[Connection], fmt: str):
+    connections_json = {}
+    for conn in conns:
+        connections_json[conn.conn_id] = {
+            'conn_type': conn.conn_type,
+            'host': conn.host,
+            'port': conn.port,
+            'is_encrypted': conn.is_encrypted,
+            'is_extra_encrypted': conn.is_encrypted,
+            'extra': json.loads(conn.extra) if conn.extra else None,
+        }
+
+    if fmt == 'yaml':
+        return yaml.dump(connections_json)
+
+    if not fmt or fmt == 'json':
+        return json.dumps(connections_json)
+
+
+def connections_export(args):
+    """Exports all connections to a file"""
+    allowed_formats = ['yaml', 'json']
+    with create_session() as session:
+        if not args.output_file:
+            print(
+                "You must pass --output-file option",
+                file=sys.stderr
+            )
+            sys.exit(1)
+        if not args.format:
+            args.format = 'json'
```

Comment:
Have you thought to try to automatically guess the format based on the file extension?